feature-libs/cart/base/components/cart-coupon/cart-coupon.component.html

```diff
@@ -7,34 +7,58 @@
     <form (ngSubmit)="applyVoucher()" [formGroup]="couponForm">
       <label class="cx-cart-coupon-title">
         {{ 'voucher.coupon' | cxTranslate }}
-        <abbr
-          *cxFeature="'a11yRequiredAsterisks'"
-          class="text-decoration-none"
-          title="{{ 'common.required' | cxTranslate }}"
-          >*</abbr
-        >
+        <ng-container *cxFeature="'!a11yDisabledButtonInsteadOfRequiredFields'">
+          <abbr
+            *cxFeature="'a11yRequiredAsterisks'"
+            class="text-decoration-none"
+            title="{{ 'common.required' | cxTranslate }}"
+            >*</abbr
+          >
+        </ng-container>
       </label>
 
       <div class="cx-cart-coupon-container">
-        <input
```

Comment:
We aren't really doing any DOM changes here but rather changing a few attribute values and removing the <cx-form-errors> tag. Should we use the `a11yDisabledButtonInsteadOfRequiredFields` flag in those attributes and with `*ngIf` on cx-form-errors instead to make this a bit cleaner? I don't mind if you do this or not since it will be cleaned up next major anyway.

Reply:
I'm not sure I understand exactly what we should do instead. If your propose is to not wrap it with the feature flag at all, then I have some concerns:

- Seems like changing attributes, like removing `required` is a breaking change. Customer could have applied styles for example by attribute selector.
- `cx-from-errors` in the case when toggle is `true` will do nothing, because the input will not have any validators applied. So it's redundant element in the DOM

Reply:
Ah ok, so this comment is mainly about having easier to read changes. Since we are changing attribute values, we could instead change the properties instead and include the flags in those. This would save us seeing alternative DOM versions with *cxFeature . So like:
```
<button
   ...
   [required]="isFeatureEnabled ? null : true
   ...
>
```
Its not critical to change and may even have issues I didn't consider. I just think it might be cleaner and would love to know what you think :slightly_smiling_face: